Sources/StreamChatUI/ChatChannelList/ChatChannelListVC.swift

```diff
@@ -417,16 +418,22 @@ open class ChatChannelListVC: _ViewController,
     // MARK: - DataControllerStateDelegate
 
     open func controller(_ controller: DataController, didChangeState state: DataController.State) {
+        handleStateChanges(state)
+    }
+
+    /// Called whenever the channels data changes or the controller.state changes.
+    /// It controls the visibility of the channel list state views.
+    open func handleStateChanges(_ newState: DataController.State) {
```

Comment:
I am not sure this one should be public. Any reason I am missing here?

Reply:
The reason is that before the customer could override the `didCHangeState` delegate event and it was enough to customize the channel state view changes. Now we call this in 2 places. So we need to extract this to a new method so that the customer can customize the state changes only in one place 👍

Reply:
Resolving since we discussed it internally ✅